erigontech/silkworm
# infra: global log level for tests, improve make test

* remove SetLogVerbosityGuard from tests
  * default to NONE level in tests (set via Settings struct default value)
  * default to INFO level in CLI tools (set via add_logging_options)
* GrpcLogGuard: default to silkworm, never restore a default GRPC logger (produces sporadic logs in tests)
* improve run_smoke_tests:
  * silent smoke tests --help output
  * extend smoke tests list and remove unit tests from it
* improve run_unit_tests:
  * use SILKWORM_TEST_SKIP instead of [ignore] and SKIP (yields cleaner output)
  * use the same rng-seed for all test commands
  * prettier run_unit_tests output
  * improve output on crashes (see https://github.com/erigontech/silkworm/pull/2500)
* enable logging of slow unit tests (see https://github.com/erigontech/silkworm/issues/2502, https://github.com/erigontech/silkworm/issues/2504)
